lambdas/service/app.py

```diff
@@ -797,6 +800,14 @@ def validate_manifest_format(format_: str):
                       f'catalog {app.catalog}. Must be one of {supported_formats}')
 
 
+def validate_version(version: str):
+    try:
+        parse_dcp2_datetime(version)
+    except ValueError:
+        raise BRE('Invalid value for parameter `version`. '
+                  'Must be an ISO 8601 formatted UTC datetime')
```

Comment:
This is misleading. Not every ISO datetime satisfies the additional requirements imposed by the DCP2 format.